openmc/source.py

```diff
@@ -198,6 +198,43 @@ def to_xml_element(self):
             element.append(self.time.to_xml_element('time'))
         return element
 
+    def plot_energy(self, n_samples=10000, seed=None, axes=None):
+        """Produce a histogram plot of source particle energy.
+
+        Parameters
+        ----------
+        n_samples : int
+            Number of sampled values to generate
+        seed : int or None
+            Initial random number seed.
+        axes : matplotlib.axes.Axes, optional
+            Axes for plot
+
+        Returns
+        -------
+        axes : matplotlib.axes.Axes
+            Axes for plot
+
+        """
+
+        energy = self.energy.sample(n_samples=n_samples, seed=seed)
+
+        import matplotlib.pyplot as plt
+
+        # Setup axes is one wasn't passed
+        if axes is None:
+            fig = plt.figure()
+            ax = plt.axes()
+            ax.set_xlabel('Energy [eV]')
+            ax.set_ylabel('Frequency')
+        else:
+            ax = axes
+
+        ax.hist(energy, bins='auto')
+
+        return ax
```

Comment:
I would return the output of `hist()` here. Something like:

```suggestion
        hist = ax.hist(energy, bins='auto')

        return hist
```

Reply:
This is just so that users have control over the objects of the histogram. It adds flexibility:
```python
import matplotlib.pyplot as plt

n, bins, patches = my_source.plot_energy()

for i, p in enumerate(patches):
    # change transparency, cause why not?
    p.set_alpha((i/len(patches)))
```
![image](https://user-images.githubusercontent.com/40028739/177590844-0adf2ed1-9a2d-4447-89a0-bfbf2f8d7159.png)

Reply:
Nice example of why hist should be returned, perhaps we should change tracks.py sometime as well
https://github.com/openmc-dev/openmc/blob/6549067c43bb00750fe4f3d772795217575380ea/openmc/tracks.py#L179